api/src/main/java/org/openmrs/module/pihcore/apploader/CustomAppLoaderFactory.java

```diff
@@ -591,25 +591,77 @@ private void enableCheckIn(Config config) {
+            } */
         }
 
-		// check-in form that appears on visit and clinicial dashboard after a visit has been started as a "Visit Action"
-        extensions.add(visitAction(CustomAppLoaderConstants.Extensions.CHECK_IN_VISIT_ACTION,
-                "mirebalais.task.checkin.label",
-                "fas fa-fw icon-check-in",
-                "link",
-                enterSimpleHtmlFormLink(PihCoreUtil.getFormResource("checkin.xml")),
-                "Task: mirebalais.checkinForm",
-                sessionLocationHasTag("Check-In Location")));
+        if (config.getCountry().equals(ConfigDescriptor.Country.SIERRA_LEONE) ) {
+            // Default SL check-in form that appears on visit and clinicial dashboard after a visit has been started as a "Visit Action"
+            extensions.add(visitAction(CustomAppLoaderConstants.Extensions.CHECK_IN_VISIT_ACTION,
+                    "mirebalais.task.checkin.label",
+                    "fas fa-fw icon-check-in",
+                    "link",
+                    enterSimpleHtmlFormLink(PihCoreUtil.getFormResource("checkin.xml")),
+                    "Task: mirebalais.checkinForm",
+                    sessionLocationHasTag("Check-In Location")));
 
-		// check-in form that appears on the Registration Page as a "Registration Action" and starts a visit
-        extensions.add(overallRegistrationAction(CustomAppLoaderConstants.Extensions.CHECK_IN_REGISTRATION_ACTION,
-                "mirebalais.task.checkin.label",
-                "fas fa-fw icon-check-in",
-                "link",
-                enterSimpleHtmlFormLink(PihCoreUtil.getFormResource("liveCheckin.xml")) + andCreateVisit(),
-                "Task: mirebalais.checkinForm",
-                sessionLocationHasTag("Check-In Location")));
+            // check-in form that appears on the Registration Page as a "Registration Action" and starts a visit
+            extensions.add(overallRegistrationAction(CustomAppLoaderConstants.Extensions.CHECK_IN_REGISTRATION_ACTION,
+                    "mirebalais.task.checkin.label",
+                    "fas fa-fw icon-check-in",
+                    "link",
+                    enterSimpleHtmlFormLink(PihCoreUtil.getFormResource("liveCheckin.xml")) + andCreateVisit(),
+                    "Task: mirebalais.checkinForm",
+                    sessionLocationHasTag("Check-In Location")));
+
+            // Maternal check-in form that appears on visit and clinicial dashboard after a visit has been started as a "Visit Action"
+            extensions.add(visitAction(CustomAppLoaderConstants.Extensions.CHECK_IN_MATERNAL_VISIT_ACTION,
+                    "mirebalais.maternalCheckin.label",
+                    "fas fa-fw icon-check-in",
+                    "link",
+                    enterSimpleHtmlFormLink(PihCoreUtil.getFormResource("checkinMaternal.xml")),
+                    "Task: mirebalais.checkinForm",
+                    sessionLocationHasTag("Check-In Maternal Location")));
```

Comment:
We also might be able to just use the "Maternal and Child Location" we use for the other MCOE forms here?  Or just the basic Check-In tag?  Might make sense to just start with one of these two until we confirm that we need a custom location set for this form?

Reply:
Right @mogoodrich .  I think we need a separate Check-In Maternal Location to handle the 2 checkin locations with different checkin encounter/forms for MCOE.  I'm fine with just the live form at MCOE (and not the live + retro).  I can simplify the code and you can review again?  Or if you can muster to energy to talk to me?

Reply:
That all sounds correct to me @lnball !